src/java/za/co/mip/ablduck/ABLDuck.java

```diff
@@ -0,0 +1,417 @@
+                        
+            if (nextClass != null) {
+                for (MemberObject member : nextClass.members) {
+                    if (member.owner.equals(inherits) && (member.meta.isPrivate == null || !member.meta.isPrivate)) 
+                        result.addInheritedmember(member);
+                }
+            
+                determineClassHierarchy(nextClass, result);
+            }
+        }
+        return result;
+    }
+
+    private void extractTemplateDirectory(File outputDir) throws IOException {
+ 
+        File template = exportResource("ablduck.zip", outputDir);
+
+        unzip(template.getAbsolutePath(), outputDir.getAbsolutePath());
+
+        template.delete();
+ 
+    }
+
+    public File exportResource(String resourceName, File outputDir) throws IOException {
+        InputStream stream = null;
+        OutputStream resStreamOut = null;
+        File resource;
+
+        try {
+            stream = ABLDuck.class.getResourceAsStream("resources/" + resourceName); 
+            if(stream == null) {
+                throw new IOException("Cannot get resource \"" + resourceName + "\" from Jar file.");
+            }
+    
+            int readBytes;
+            byte[] buffer = new byte[BUFFER_SIZE];
+            
+            resource = new File(outputDir, resourceName); 
+            resStreamOut = new FileOutputStream(resource.getAbsolutePath());
+            while ((readBytes = stream.read(buffer)) > 0) {
+                resStreamOut.write(buffer, 0, readBytes);
+            }
+        } catch (IOException ex) {
+            throw ex;
+        } finally {
+            if(stream != null)
+                stream.close();
+
+            if(resStreamOut != null)
+                resStreamOut.close();
+        }
+    
+        return resource;
+    }
+    
+    public void unzip(String zipFilePath, String destDirectory) throws IOException {
+        File destDir = new File(destDirectory);
+        if (!destDir.exists()) {
+            destDir.mkdir();
+        }
+        ZipInputStream zipIn = new ZipInputStream(new FileInputStream(zipFilePath));
+        ZipEntry entry = zipIn.getNextEntry();
+        // iterates over entries in the zip file
+        while (entry != null) {
+            String filePath = Paths.get(destDirectory, entry.getName()).toString();
+            if (!entry.isDirectory()) {
+                // if the entry is a file, extracts it
+                extractFile(zipIn, filePath);
+            } else {
+                // if the entry is a directory, make the directory
+                File dir = new File(filePath);
+                dir.mkdir();
+            }
+            zipIn.closeEntry();
+            entry = zipIn.getNextEntry();
+        }
+        zipIn.close();
+    }
+    
+    private static void extractFile(ZipInputStream zipIn, String filePath) throws IOException {
+        BufferedOutputStream bos = new BufferedOutputStream(new FileOutputStream(filePath));

```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Close this "FileOutputStream" in a "finally" clause. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS2095)